src/components/cc-button/cc-button.js

```diff
@@ -115,6 +119,40 @@ export class CcButton extends LitElement {
     this._cancelMode = false;
   }
 
+  /**
+   * The `aria-label` attribute should only be set if the `accessibleName` prop is set or if the button only shows an image with no text.
+   *
+   * @returns {string|undefined} the value of the `aria-label` attribute or `undefined` if the attribute should not be set.
+   */
+  _getAriaLabel () {
+    if (this.accessibleName != null) {
+      return this.accessibleName?.trim() ?? '';
+    }
+
+    if (this.hideText && this.image != null) {
+      return this.textContent?.trim() ?? '';
+    }
+
+    return undefined;
+  }
+
+  /**
+   * The `title` attribute should only be set if the `accessibleName` prop is set or if the button only shows an image with no text.
+   *
+   * @returns {string|undefined} the value of the `title` attribute or `undefined` if the attribute should not be set.
+   */
+  _getTitle () {
+    if (this.accessibleName != null) {
+      return this.accessibleName?.trim() ?? '';
+    }
+
+    if (this.hideText && this.image != null) {
+      return this.textContent?.trim() ?? '';
```

Comment:
I don't think `this.textContent` can be null but let's keep this `?.` to be sure for now.

Reply:
Eh indeed, it should be an empty string if there's no text content, and there shouldn't be cases with no text content cause it's bad for a11y anyway :smile: 

I'll remove it :+1: